### PR TITLE
Add Blog Subscription Feature

### PR DESCRIPTION
**Pull request by michellevit**
## TLDR
This PR introduces a new subscription feature to the website, enabling visitors to subscribe for email notifications whenever a new blog post is published. The feature leverages a Cloudflare Worker to handle API requests and integrates with Mailchimp for managing subscriptions and sending notifications, but requires some additional setup for full functionality.

### RSS Generator Script
- Implemented a script (`generate-rss.js`) to generate an RSS feed during the build for non-hidden blog posts.

### Subscribe Component
- Created a Subscribe component (`Subscribe.jsx`) that allows users to submit their email addresses and first names to sign up for blog updates.
- Integrated this component across relevant pages of the website for easy user accessibility.

### Cloudflare Worker
- Developed code for a Cloudflare Worker (`subscribe-worker.js`) to handle API requests to Mailchimp.

### README Updates
- Updated the README to include comprehensive setup instructions for configuring the website on Cloudflare and setting up the Mailchimp RSS-to-email campaign.

### Assumptions
- The website is hosted on Cloudflare using the free tier plan.
- Deployment is managed through Cloudflare Pages.
- The domain uses Cloudflare nameservers.

### Setup Required
- Additional configuration steps are necessary for the subscription feature to function correctly:
  - Configuration of a Cloudflare Worker.
  - Setup of a Mailchimp RSS-to-email campaign.
- Detailed instructions are provided in the README under 'How to Set Up the Blog Mailing List'.

### Why a Cloudflare Worker?
- The Cloudflare Pages free tier does not support traditional server-based backends, but the  Cloudflare Workers feature provides an alternative to this limitation.

### Estimated Setup Time
- The setup process, including Mailchimp configuration and Cloudflare Worker deployment, is expected to take less than 30 minutes.

### Date Metadata in Blog Posts
- To ensure that blog posts trigger Mailchimp's RSS-to-email campaigns correctly and timely:
  - The 'date' field in each blog post must accurately reflect the date and time of publication using the format `YYYY-MM-DDTHH:MM:SS-08:00` for Pacific Standard Time (*'-07:00' for PDT).
  - Example: For a post meant to be published at 12:30 PM PST on May 22, 2024, the date field should be set as `2024-05-22T12:30:00-08:00`.
  - Posts with dates set in the past relative to the most recent Mailchimp campaign check will not trigger a new email notification, as Mailchimp uses this date to determine the novelty of the content.

## Testing
- This feature has been implemented and tested on my personal website: [michelleflandin.com](http://michelleflandin.com). While this site serves as a visual demo to showcase the Subscribe component's user interface, please note that the actual subscription functionality is not active due to hardcoded URLs in the backend configuration.
- This implementation will remain accessible as a live demo until the closure of this pull request.
- Compatibility and responsiveness have been verified across multiple browsers including Chrome, Firefox, Safari, and Edge.

## Request for Review

I invite you to review the implemented changes and provide feedback. Should there be any questions or suggestions for further adjustments, please feel free to reach out. Your input is highly valued and appreciated. Thank you for your consideration!

### CODE REVIEW

File: scripts/subscribe-worker.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                });
+            }
+  
+            const data = {
+                email_address: email,
+                status: 'subscribed',
+                merge_fields: { FNAME: firstName },
+            };
+  
+            const DATACENTER = MAILCHIMP_API_KEY.split('-')[1];
```
Comment (paxsonsa) [inline scripts/subscribe-worker.js:31]: shouldn't this reference the env?

File: scripts/subscribe-worker.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  
+            const url = `https://${DATACENTER}.api.mailchimp.com/3.0/lists/${MAILCHIMP_AUDIENCE_ID}/members`;
```
Comment (paxsonsa) [inline scripts/subscribe-worker.js:33]: shouldn't the `MAILCHIMP_AUDIENCE_ID` reference the process env?

File: scripts/generate-rss.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const fs = require('fs');
+const path = require('path');
+const RSS = require('rss');
+const matter = require('gray-matter');
+
+const siteUrl = 'https://andrewpaxson.com'
```
Comment (paxsonsa) [inline scripts/generate-rss.js:6]: we should probably reference this from the env no?

File: README.md [hunk 4]
```diff
@@ -19,3 +19,132 @@ This is the repo for my personal website.
...
+- Setup a Mailchimp RSS Campaign:
+  - Signup/login to MailChimp
+    - Optional: Change profile timezone: 
+      - Select the profile icon
+      - Select Profile 
+      - Go to the 'Settings' dropdown and select 'Details'
+      - Update the Timezone field (e.g., '(GMT-07:00) Vancouver')
+      - Select 'Save' 
+    -  Create an RSS campaign: 
+      - Go to this [MailChimp URL](https://us22.admin.mailchimp.com/campaigns/#/create-campaign/explore/rss)
```
Comment (paxsonsa) [inline README.md:91]: Probably avoid a cdn url :)

**Comment by paxsonsa** [comment]
@michellevit This is such a nice gesture thanks. I have to ask what motivated you to do this and did you mean to submit a PR lol

File: components/Subscribe.jsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+          className="w-full bg-sky-500 text-white px-3 py-2 rounded-md shadow hover:bg-sky-600 dark:bg-sky-400 dark:hover:bg-sky-500"
+        >
+          Subscribe
+        </button>
+        {status === 'success' && <p className="text-green-500 text-center">Subscribed successfully!</p>}
+        {status === 'error' && <p className="text-red-500 text-center">Subscription failed. Please try again.</p>}
+      </form>
+    </div>
+  );
+};
```
Comment (paxsonsa) [inline components/Subscribe.jsx:77]: I would definitely update this styling to match my page but thanks for go initial implementation!

**Review by paxsonsa — CHANGES_REQUESTED** [review]
